File: new_files/s03/part2/device_control.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/types.h>
#include <fcntl.h>
#include <unistd.h>
#include <string.h>

int main(int argc, char **argv)
{
    char readBuffer[1024] = {0};
    int readFile=open("/dev/iut_device", O_RDWR);
    char buffer[100]={0};
    sprintf(buffer, "%s,%s,%s,%s", argv[1], argv[2], argv[3], argv[4]);
    printf("%s\n", buffer);
    write(readFile, buffer, 100);
    read(readFile,readBuffer, 1024);
    fprintf(stdout,"%s\n",readBuffer);
    close(readFile);

    return 0;
}

File: new_files/s03/part2/device_control.py
#!/bin/python3

import os
import sys

def read_from_file(fd: int):
    init_data = os.read(fd, 1024)
    print(init_data.decode())

def write_to_file(fd: int, cmd: str):
    os.write(fd, cmd.encode())

def main():
    fd = os.open("/dev/iut_device", os.O_RDWR)
    buffer = sys.argv[1] + "," + sys.argv[2] + "," + sys.argv[3] + "," + sys.argv[4]
    print(buffer)
    write_to_file(fd, buffer)
    read_from_file(fd)
    os.close(fd)

if __name__ == "__main__":
    main()
